core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -90,6 +91,11 @@
     private final int minimumNumberOfDrivers;
     private final int guaranteedNumberOfDriversPerTask;
     private final int maximumNumberOfDriversPerTask;
+
+    @GuardedBy("this")
```

Comment:
this is not `@GuardedBy("this")` as it's a bean that can be accessed anynchroniously

Reply:
You right.